src/pages/groupComparison/GroupComparisonURLWrapper.ts

```diff
@@ -0,0 +1,33 @@
+import URLWrapper from "../../shared/lib/URLWrapper";
+import ExtendedRouterStore from "../../shared/lib/ExtendedRouterStore";
+import {computed} from "mobx";
+import {getTabId, GroupComparisonTab} from "./GroupComparisonUtils";
+import autobind from "autobind-decorator";
+import {OverlapStrategy} from "./GroupComparisonStore";
+
+export type GroupComparisonURLQuery = {
+    sessionId: string;
+    groupOrder?:string; // json stringified array of names
+    unselectedGroups?:string; // json stringified array of names
+    overlapStrategy?:OverlapStrategy;
+};
+
+export default class GroupComparisonURLWrapper extends URLWrapper<GroupComparisonURLQuery> {
+    constructor(routing:ExtendedRouterStore) {
+        super(routing, [
+            { name:"sessionId", persisted: false },
+            { name:"groupOrder", persisted: false },
+            { name:"unselectedGroups", persisted: false },
+            { name:"overlapStrategy", persisted: false }
```

Comment:
are these typesafe?  I mean, will ts complain if these are not represented in GroupComparisonURLQuery?

Reply:
yeah - setting the generic parameter with `URLWrapper<GroupComparisonURLQuery>` means that `name` must be `keyof GroupComparisonURLQuery` (see `URLWrapper.ts::Property`